### PR TITLE
@W-15803283@ -  BREAKING CHANGE: Migrating from sfdx to sf 

### PR DESCRIPTION
**Pull request by BharathAdhikari**
### What does this PR do?
Migrating from the SFDX CLI to SF

### What issues does this PR fix or reference?
#<Insert GitHub Issue>, [W-15803283](https://gus.lightning.force.com/a07EE00001rwFaeYAE)

### Functionality Before
Was able to create a scratch org and sample store on the org created.

### Functionality After
we can create a scratch org and sample store on the org created.

### How to Test/Testing Effort 
Verified the scratch org creation locally after updating the commands
Screenshot 2024-05-27 at 3 22 40 PM
For store creation, there is an existing issue with the schema where we were unable to deploy the retrieved digital experience bundles due to a schema issue. Will verify this flow once the issue is fixed.
Verified the e2e changes by testing with 250(code line) 
Screenshot 2024-06-04 at 1 25 16 AM

### CODE REVIEW

File: test/lib/utils/shell.test.ts [hunk 1]
```diff
@@ -8,16 +8,16 @@
...
 //     });
 // });
 // describe('Shell Json Sfdx', () => {
 //     it('should get api version in an object', async function () {
 //         this.timeout(20000);
-//         const res = shellJsonSfdx<{ apiVersion }>('sfdx force');
+//         const res = shellJsonSfdx<{ apiVersion }>('sf force');
 //         assert.ok(res.result.apiVersion);
 //     });
 // });
```
Comment (tarcang) [inline test/lib/utils/shell.test.ts:23]: Just delete this test, since it is commented out already.

**Review by tarcang — COMMENTED** [review]
Overall, great work! I just have some comments.

File: test/lib/utils/shell.test.ts (hunk 1 above)
Comment (BharathAdhikari) [reply test/lib/utils/shell.test.ts:23]: Sure, will remove it.

File: src/lib/utils/args/flagsUtils.ts [hunk 2]
```diff
@@ -136,7 +136,7 @@ export function appendCommonFlags(cmd: string, flags: OutputFlags<any>, logger:
     if (flags?.apiversion && cmd.startsWith('sf ')) {
         cmd = `${cmd} --api-version=${flags.apiversion as string}`;
     } else if (flags?.apiversion) {
-        cmd = `${cmd} --apiversion=${flags.apiversion as string}`;
+        cmd = `${cmd} --api-version=${flags.apiversion as string}`;
```
[the remark this answers is not in the document]
Comment (BharathAdhikari) [reply src/lib/utils/args/flagsUtils.ts:139]: Sure, just kept it till we update the Shane command as it is still using sfdx

File: README.md [hunk 3]
```diff
@@ -362,12 +362,12 @@ OPTIONS
                                                                                     this command invocation
 
 EXAMPLES
-  sfdx commerce:files:copy -y --copySourcePath "~/myexamplefilesdirectory" --filestocopy "file1.txt,file2.txt"
+  sf commerce:files:copy -y --copySourcePath "~/myexamplefilesdirectory" --filestocopy "file1.txt,file2.txt"
```
[the remark this answers is not in the document]
Comment (BharathAdhikari) [reply README.md:365]: yes, but as the sf command is taking care of this conversion haven't updated the commands as we might have to also update the structure at the file level.

File: src/commands/commerce/products/import.ts [hunk 4]
```diff
@@ -82,15 +82,13 @@ export class ProductsImport extends SfdxCommand {
...
-                            this.flags['products-file-csv'] as string
-                        } -u "${this.org.getUsername()}" --json`,
-                        this.flags,
-                        this.logger
-                    )
-                );
+                const productsFileCsv: string = this.flags['products-file-csv'] as string;
+                const username: string = this.org.getUsername();
+
+                const command = `sfdx shane:data:file:upload -f ${productsFileCsv} -u ${username} --json`;
```
[the remark this answers is not in the document]
Comment (BharathAdhikari) [reply src/commands/commerce/products/import.ts]: updated the sf equivalent implementation.

**Comment by BharathAdhikari** [comment]
@tarcang  as discussed, verified the e2e pipeline by adding 250(code-line) devhub credentials. Attaching the e2e pipeline run status for ref :
Screenshot 2024-06-04 at 1 25 16 AM

File: docs/setup-guide.md [hunk 5]
```diff
@@ -12,101 +12,95 @@
 
 1. Install Salesforce CLI. You can install the CLI with either npm or with a downloadable installer for your specific operating system. See [Salesforce CLI Setup Guide](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm) for more information.
     ```bash
-    npm install --global sfdx-cli
+    npm install @salesforce/cli@2.43.7 --global
```
Comment (tarcang) [inline docs/setup-guide.md:15]: They don't have to enter a version name.

File: docs/setup-guide.md [hunk 6]
```diff
@@ -12,101 +12,95 @@
...
-    # sfdx commerce:store:create -n b2bstore02 -o b2b -b b2b_buyer@1commerce.com -u demo_org_1@1commerce.com -v devhubtest1sdb3@mydevhub.com --apiversion=57.0
+    # sf commerce:store:create -n b2bstore02 -o b2b -b b2b_buyer@1commerce.com -u demo_org_1@1commerce.com -v devhubtest1sdb3@mydevhub.com --apiversion=57.0
     ```
 
-10. Create a B2C LWR store.
+9. Create a B2C LWR store.
 
     ```bash
-    sfdx commerce:store:create -n <<STORE_NAME>> -o b2c -b <<BUYER_USER_EMAIL>> -u <<ORG_USERNAME>> -v <<DEVHUB_USERNAME>> --apiversion=<<API_VERSION>>
+    sf commerce:store:create -n <<STORE_NAME>> -o b2c -b <<BUYER_USER_EMAIL>> -u <<ORG_USERNAME>> -v <<DEVHUB_USERNAME>> --apiversion=<<API_VERSION>>
```
Comment (tarcang) [inline docs/setup-guide.md:87]: Please make sure that using sf with colons still work.

**Review by tarcang — APPROVED** [review]
Overall looks good! Please review the comments.

File: docs/setup-guide.md (hunk 5 above)
Comment (BharathAdhikari) [reply docs/setup-guide.md:15]: The import functionality was pushed as a part of 2.43.7 itself and as the pipelines try to install 2.42.x. the new SF import command fails, Hence added with the specific version.

Will  remove this before moving this to main/release.

File: docs/setup-guide.md (hunk 6 above)
Comment (BharathAdhikari) [reply docs/setup-guide.md:87]: confirmed this from here: https://oclif.io/docs/topic_separator, But was not able to verify this locally.

**Pull request merged**